includes/amp-helper-functions.php

```diff
@@ -899,6 +935,33 @@ function amp_get_content_sanitizers( $post = null ) {
 	 */
 	$sanitizers = apply_filters( 'amp_content_sanitizers', $sanitizers, $post );
 
+	if ( amp_is_dev_mode() ) {
```

Comment:
I think it would be preferable to have this piece of logic be part of the `AMP_Dev_Mode_Sanitizer` class instead, and then make that sanitizer decide when it acts or not, instead of hardcoding it in here.

It should be the responsibility of the `AMP_Dev_Mode_Sanitizer` to know when to do something and what to do, not some helper function collection.

Reply:
I did think about that, but I think it's preferable here because this function is inherently connected directly with WordPress as it applies a filter. There are two reasons for why this should not be in the sanitizer:

1. It is important that the array of sanitizer args returned by `amp_get_content_sanitizers()` be deterministic in what the resulting sanitized document be. In other words, we cannot apply filters _inside_ of a sanitizer because then we cannot use the sanitizer args as a source of truth for computing a cache key for the post-processed output, as we can now.
1. We should be trending the sanitizers toward being pure PHP classes that are do not connect with WordPress APIs (especially the plugin API per above) as this is the direction needed for abstracting the sanitizers onto a CMS-agnostic package.

Additionally, this `amp_is_dev_mode()` function would be useful outside of the context of a sanitizer. For example, a theme/plugin may want to call this function during template generation to determine whether it can output some dev code (e.g. Query Monitor).

The `amp_is_dev_mode()` is gatekeeping whether or not the `AMP_Dev_Mode_Sanitizer` should be used in the first place. So it is operating at a higher level.

Aside: should it rather be named `is_amp_dev_mode()`?

Reply:
> Aside: should it rather be named is_amp_dev_mode()?

I'm seeing the `amp_` as a prefix here, as this could just as well be used outside of the AMP plugin, so I think the name is fine.